src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,342 @@
+			if (intentAction is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:GetIconId, ResolveActivity is null");
+			}
+
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var childToast = xmlDoc.GetElementsByTagName("toast");
+			if (childToast.Count == 1)
+			{
+				var childLaunch = childToast[0]?.Attributes?.GetNamedItem("launch");
+				if (childLaunch != null)
+				{
+					var launchValue = childLaunch.Value;
+					if (launchValue is null)
+					{
+						throw new Exception("It cannot happen, but to keep compiler happy...");
+					}
+					toastArgument = launchValue;
+				}
+			}
+
+			intentAction.PutExtra("Uno.internal.ToastArgument", toastArgument);
+
+			// we should have intent.data field different for different toast arguments,
+			// else - next intent would overwrite previous.
+			// it can be any URI, e.g. http://dummy.com/dummy?arg=  
+			var data = Android.Net.Uri.Parse("http://unotask.com/toast?arg=" + Android.Net.Uri.Encode(toastArgument));
+			intentAction.SetData(data);
+
+			// 12345: arbitrary number, it can be any value (as it is not used in Uno code anywhere)
+			var retVal = Android.App.PendingIntent.GetActivity(Android.App.Application.Context, 12345, intentAction, 0); // 12345=requestCode
+			if (retVal is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:CreateToastIntent, GetActivity is null");
+			}
+
+			return retVal;
+		}
+
+
+		public void Show(ToastNotification notification)
+		{
+			var androidNotification = new Android.App.Notification();
+			androidNotification.Category = Android.App.Notification.CategoryMessage;
+
+			Android.App.Notification.Builder builder = GetToastBuilder();
+
+			// we use XmlDocument for retrieving text items and 'launch' attribute
+			var xmlDoc = new XmlDocument();
+			xmlDoc.LoadXml(notification.Content.GetXml());
+
+			// packageName is used in GetIconId and to create launch intent
+			var context = Android.App.Application.Context;
+			if (context.PackageName is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:Show, Context is null");
+			}
+
+			string packageName = context.PackageName;
+
+			// extract <text> nodes from XML 
+			SetToastTexts(builder, xmlDoc);
+
+			// we have to set icon ("This is the only user-visible content that's required", says doc).
+			int iconId = GetIconId(packageName);
+			if (iconId == 0)
+			{
+				throw new InvalidOperationException("ToastNotifier: Unable to get application default Icon, make sure to set one in Android manifest file (application/@android:icon)");

```

Comment:
Two things:
- Let's add this particular explanation in the documentation markdown file.
- Could you modify the manifest to add the existing icon in the resources? (for both SamplesApp.Droid and SamplesApp.netcoremobile)

Reply:
First part: done.

Second part, about icon. Do we have icon for app, or should I create one?

Reply:
Do we have icon for app, or should I create one?

Reply:
Thanks for the updates! Let's use [this one](https://github.com/unoplatform/uno/blob/master/src/SamplesApp/SamplesApp.Shared/Assets/storelogo.png).

Reply:
Maybe done :)